patch-cluster-node.sh

```diff
@@ -41,35 +48,54 @@ do
             ;;
         R)
             ROLLBACK=1
+            if [ -z ${OPTARG} ]; then
+                echo "Exiting; Backup version parameter is required for rollback"
+                exit 4
+            fi
+            BACKUP_SUFFIX=".original_${OPTARG}"
             ;;
         *|?|h)
             echo "Usage: $0 [-a] [-u HDFS_USER] [-t TARGET_VERSION] [-p DIRECTORY_PREFIX] [-P HDFS_DIRECTORY_PREFIX] [-R] [-?]"
             echo ""
             echo "Where:"
             echo "  -a              Update HDFS contents. This switch should only be specified for ONE node in a cluster patch."
             echo "  -u HDFS_USER    Specifies the user name with superuser privileges on HDFS. Applicable only if the -a switch is specified."
-            echo "  -t TARGET_VERSION "
-            echo "                  Specifies a Release name in the associated Github repo. This release will contain the .jar file to patch."
             echo "  -p DIRECTORY_PREFIX "
             echo "                  Specifies a prefix that is specific to the Hadoop distro & version to search for files to patch."
             echo "  -P HDFS_DIRECTORY_PREFIX "
             echo "                  Specifies a prefix that is specific to the Hadoop distro & version to search on HDFS for files to patch."
             echo "  -R              Rollback installation. Restores previously backed up versions of hadoop-azure jar file. Rollback for HDFS "
-            echo "                  should follow same model as deployment."
+            echo "                  should follow same model as deployment. Specify the backup version for the rollback. Ex: Specify 2020-06-07-10-10-10 "
+            echo "                  for the backup file named hadoop-azure.*.jar.original_2020-06-07-10-10-10"
             exit 1
             ;;
     esac
 done
 
-which jq > /dev/null
-if [ $? -ne 0 ]; then
 
-    echo "This script requires jq to run. Please install using preferred package manager"
-    exit 1
-fi
+[[ "${DIR_PREFIX}" != */ ]] && DIR_PREFIX="${DIR_PREFIX}/"
```

Comment:
Not sure this is a change we want. Relying on user to correctly specify the path prefixes degrades the usability of the script and only opens an opportunity for error. 
Consider either hard-coding default values or retaining the properties file mechanism.

Reply:
Currently planning to go ahead with hard coding.